internal/objects/deployment/deployment.go

```diff
@@ -304,7 +304,6 @@ func DesiredDeployment(contour *operatorv1alpha1.Contour, image string) *appsv1.
 					ServiceAccountName:            objutil.ContourRbacName,
 					RestartPolicy:                 corev1.RestartPolicyAlways,
 					SchedulerName:                 "default-scheduler",
-					SecurityContext:               objutil.NewUnprivilegedPodSecurity(),
```

Comment:
I think we don't need to move this line.
If `contour.ContourSecurityContextExists()` is `true`, we just overrides `deploy.Spec.Template.Spec.SecurityContext`.

Reply:
It might be premature optimization but I was hoping to avoid an unnecessary allocation.